### ISSUE TITLE
OSError: [WinError 6] The handle is invalid

### ISSUE THREAD

**Issue by joaompinto** [prompt-toolkit/python-prompt-toolkit#1433]
Running on Windows 10

$ python -V
Python 3.8.4

$ ptpython -V
3.0.17

$ ptpython

print("ok")
Cancelling an overlapped future failed
future: <_OverlappedFuture pending overlapped=<pending, 0x235e16bedc0> cb=[BaseProactorEventLoop._loop_self_reading()]>
Traceback (most recent call last):
File "c:\users\lameg\appdata\local\programs\python\python38\lib\asyncio\windows_events.py", line 66, in _cancel_overlapped
self._ov.cancel()
OSError: [WinError 6] The handle is invalid
ok

**Comment by wedesoft** [comment]
Yes, [joaompinto's commit](https://github.com/joaompinto/python-prompt-toolkit/commit/e7237abc0a4309d37d942b755fb49197188812c8) fixes it for me as well.

**Comment by IlyaOrson** [comment]
Could this fix get merged for the next version?

**Comment by kylepollina** [comment]
The merge request https://github.com/prompt-toolkit/python-prompt-toolkit/pull/1434 still has work to do. The maintainer wants it designed differently
